tests/python/pants_test/tasks/test_changed_target_integration.py

```diff
@@ -10,70 +10,91 @@
 
 from pants.base.build_environment import get_buildroot
 from pants.util.contextutil import temporary_dir
-from pants.util.dirutil import safe_open
+from pants.util.dirutil import fast_relpath, safe_file_dump, safe_mkdir, safe_open
 from pants_test.pants_run_integration_test import PantsRunIntegrationTest
 from pants_test.testutils.git_util import initialize_repo
 
 
 class ChangedTargetGoalsIntegrationTest(PantsRunIntegrationTest):
   @contextmanager
   def known_commits(self):
+    """Creates an anonymous git repository under the buildroot."""
```

Comment:
Why is it important for it to be under the real buildroot? I'm getting quite confused by the nesting of real buildroot, test gitroot and test buildroot.